doc/developers/js_app.rst

```diff
@@ -0,0 +1,167 @@
+JavaScript Application
+======================
+
+CCF includes a native ``js_generic`` application which can execute JavaScript applications proposed through governance.
+
+CCF's JavaScript environment is similar to that in browsers, with the following differences:
+
+- only `JavaScript modules <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules>`_ are supported,
+- no `Web APIs <https://developer.mozilla.org/en-US/docs/Web/API>`_ are available,
+- module imports cannot be URLs and must be relative paths.
+
+.. _js-app-bundle:
+
+App bundle
+----------
+
+The most basic CCF JavaScript app has the following folder structure:
+
+.. code-block:: bash
+
+    $ tree --dirsfirst my-app
+    my-app
+    ├── src
+    │   └── app.js
+    └── app.json
+
+We also call this a `JavaScript app bundle` since it is the format used for deployment.
+It consists of metadata (``app.json``) and one or more JavaScript modules (``src/``)
+-- more details in later sections.
```

Comment:
Better to link to the other sections directly? FYI, we use full paths for references, e.g.:
```
:ref:`issue new business requests <users/issue_commands:Issuing Commands>`
```

Reply:
Is there a way to reference a page itself?
This looks really ugly:
```
See the :ref:`tsoa-based app example <developers/js_app_tsoa_npm:TypeScript Application using Node.js, npm, and tsoa>` on how to generate OpenAPI definitions with TypeScript.
`` 
And it breaks if I change the page title.

Reply:
You don't have to link to the section if you just want to link to the page, but it is more precise that way. You can create named refs if you prefer too: https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html#the-ref-role

Reply:
Doesn't work if I omit the section title. I think it's a sphinx (plugin) issue/limitation when it creates the labels.

Reply:
In autoref you mean? I'll take a look.

Reply:
I think sphinx.ext.autosectionlabel